### DISCUSSION TITLE
actix_cors is always returing 404 on OPTIONS requests

### DISCUSSION THREAD

**Discussion by alvarosan** [actix/examples#581] [Q&A] [answered]
Disclaimer: I have looked around for a solution for this and I could not find anyone else having this problem so I might be missing something fundamental.

I could make actix_cors work in my server (actix_web =  "4.2.1" , actix_cors = "0.6.2").   No errors are thrown at runtime when starting the application but whenever I make an OPTIONS request to a registered endpoint I keep getting a 404 response.

After several attempts to fix this, I found your reference application (`examples/cors`) and unfortunately I could also reproduce the problem there. It is unclear what is happening (I could reproduce it both in the browser and with postman).  Here are some screenshots:

Working POST request:

![Screenshot from 2022-11-08 11-02-52](https://user-images.githubusercontent.com/767484/200537559-ae28580d-e6d3-47f3-afd2-905bb59c0174.png)

Failing OPTIONS request:
![Screenshot from 2022-11-08 11-03-14](https://user-images.githubusercontent.com/767484/200537615-5ed6f550-54b7-4a7e-9f9c-e96bc2a56b6a.png)

Let me know if you need more info, thanks.

Alvaro

**Comment by robjtede** [comment, marked as answer]
You need to send the request with an Origin header. This is what activates the CORS protocol.

**Reply by alvarosan** [reply]
Hey @robjtede , thanks for your reply.  I can confirm this was my problem, it works like a charm now.  It would probably be worth adding a note about this in the docs (for those like me not very familiar with the CORS protocol).  Thanks!
